src/NuGet.Core/NuGet.ProjectModel/CacheFileFormat.cs

```diff
@@ -81,6 +83,25 @@ private static CacheFile ReadCacheFile(JObject cursor)
             var cacheFile = new CacheFile(hash);
             cacheFile.Version = version;
             cacheFile.Success = success;
+
+            if (version >= 2)
```

Comment:
Would not this change increase the full restore execution time? I am thinking this way because the cache file read and write will be increased. If it the cache got invalidated the full restore will take the performance hit. 
If my understanding is correct do we know how much the full restore performance will gain and how much the noop perf restore will win?

Reply:
There will be a slight increase to the time it takes to write the cache file but that happens during restore which is normally OK to be a little slower.  On the bright side, I've reduced the restore time when generating these files by as much as 50% so there won't be any perceived slowness for this new operation.

Reply:
Can we quantify the changes in different types of projects?
For example how do the following perform (note, this is just a suggestion) 

1 project
10 projects
NuGet.Client sln
Roslyn sln
CBT

Reply:
I restored 2 projects with and without this change and the wall clock time is identical (~1 second)

700 projects takes 20 seconds without this change and 15 seconds with this change

Those numbers are from my locally built copies, shipped assemblies should be a little lower since they'll be Release and NGen'd in the near future.